src/test/controller_mapping_file_handler_test.cpp

```diff
@@ -0,0 +1,1006 @@
+    EXPECT_CALL(*mapping, addModule(_, _)).Times(0);
+
+    addScriptFilesToMapping(
+            doc.documentElement(),
+            mapping,
+            QDir());
+}
+
+TEST_F(LegacyControllerMappingFileHandlerTest, canParseScreenMapping) {
+    QDomDocument doc;
+    doc.setContent(QByteArray(R"EOF(
+            <controller id="DummyDevice">
+                <screens>
+                    <screen identifier="main" width="480" height="360" targetFps="20" pixelType="RBGA" splashoff="2000" />
+                </screens>
+                <scriptfiles>
+                    <file filename="DummyDeviceDefaultScreen.qml" />
+                </scriptfiles>
+                <qmllibraries>
+                    <library path="foobar/" />
+                </qmllibraries>
+            </controller>
+            )EOF"));
+
+    auto mapping = std::make_shared<MockLegacyControllerMapping>();
+    // This file always gets added
+    EXPECT_CALL(*mapping,
+            addScriptFile(QString(REQUIRED_SCRIPT_FILE),
+                    QString(""),
+                    _, // gmock seems unable to assert QFileInfo
+                    LegacyControllerMapping::ScriptFileInfo::Type::Javascript,
+                    true));
+
+    EXPECT_CALL(*mapping,
+            addScriptFile(QString("DummyDeviceDefaultScreen.qml"),
+                    QString(""),
+                    QFileInfo("/dummy/path/DummyDeviceDefaultScreen.qml"),
+                    LegacyControllerMapping::ScriptFileInfo::Type::Qml,
+                    false));
+    EXPECT_CALL(*mapping,
+            addScreenInfo(QString("main"),
+                    QSize(480, 360),
+                    20,
+                    1,
+                    std::chrono::milliseconds(2000),
+                    QImage::Format_RGBA8888,
+                    LegacyControllerMapping::ScreenInfo::ColorEndian::Little,
+                    false,
+                    false));
+    EXPECT_CALL(*mapping, addModule(QFileInfo("/dummy/path/foobar"), false));
+
+    addScriptFilesToMapping(
+            doc.documentElement(),
+            mapping,
+            QDir());
+}
+
+TEST_F(LegacyControllerMappingFileHandlerTest, screenMappingTargetFPS) {
+    QDomDocument doc;
+    doc.setContent(
+            QByteArray(R"EOF(
+            <controller id="DummyDevice">
+                <screens>
+                    <screen identifier="main" width="1" height="1" targetFps="20" />
+                </screens>
+            </controller>
+            )EOF"));
+
+    auto mapping = std::make_shared<MockLegacyControllerMapping>();
+    // This file always gets added
+    EXPECT_CALL(*mapping,
+            addScriptFile(QString(REQUIRED_SCRIPT_FILE),
+                    QString(""),
+                    _, // gmock seems unable to assert QFileInfo
+                    LegacyControllerMapping::ScriptFileInfo::Type::Javascript,
+                    true));
+    EXPECT_CALL(*mapping, addScreenInfo(_, _, 20, _, _, _, _, _, _));
```

Comment:
ahh ok I now I see what you're doing here.  Instead of the expect_call pattern, why not add an accessor like GetScreenInfo and then you can confirm that the data in the object matches the data you expect to see?  Besides, in general it's better that your tests are testing *state*, not testing *implementation*.  It's not important that a function get called with a specific parameter.  It's important that the object has a specific state when the call is complete.

so I would rewrite these tests as

```
addScriptFilesToMapping(...);
got = mapping->GetScreenInfo(); // could be a FRIEND_TEST protected function
EXPECT_EQUAL(20, got.TargetFps);
```

Reply:
I cannot do this since `addScriptFilesToMapping` is a mock, and I want to ensure the single call (`times(1)`, by default IIRC).

Turns out there is a `FieldsAre` for the matcher so you can do ` EXPECT_CALL(*mapping, addScreenInfo(FieldsAre(_, _, 20, _, _, _, _, _, _)));`

> Besides, in general it's better that your tests are testing state, not testing implementation. It's not important that a function get called with a specific parameter

This test is focusing on the file handling state (as per the name), this is why `LegacyControllerMapping` is mocked (`MockLegacyControllerMapping`). It focuses on making sure that the handler is able to properly parse and interpret mapping definition. 
Arguably, `LegacyControllerMapping` should have its own test file to ensure all accessors work as expected, but that class has a lot of members, and doesn't have any tests written, so I won't add this in this PR.